eng/pipelines/templates/steps/vcpkg.yml

```diff
@@ -1,146 +1,59 @@
-# Assumes that the VM image is in a variable called OSVmImage
-
 parameters:
   # Use the variable name itself (no $() or other wrapping syntax). This is
   # because we use runtime and macro expressions for conditions and script
   # invocations
   DependenciesVariableName: vcpkg.deps
 
-  # Ref at which to check out (can be commit SHA or tag in the form of
-  # `tags/<tag>`)
-  VcpkgVersion: tags/2020.06
-
 steps:
-  # Set VCPKG_INSTALLATION_ROOT location for cache on Mac OS hosts
-  - task: Bash@3
-    inputs:
-      targetType: inline
-      workingDirectory: $(Agent.TempDirectory)
-      script: |
-        echo "##vso[task.prependpath]$(Agent.TempDirectory)/vcpkg"
-        echo "##vso[task.setvariable variable=VCPKG_INSTALLATION_ROOT;]$(Agent.TempDirectory)/vcpkg"
-    displayName: MacOS - Set VCPKG_INSTALLATION_ROOT
+  - pwsh: |
+      $TargetPath = "$(Agent.TempDirectory)/vcpkg"
+      Remove-Item -Path $TargetPath -Recurse -Force -ErrorAction Ignore
+      New-Item -ItemType Directory -Path $TargetPath  -Force
+
+      $VcpkgCommit = $(Get-Content eng/vcpkg-commit.txt)
+      Write-Host "Target Path for vcpkg: $TargetPath"
+      Write-Host "Vcpkg SHA: $VcpkgCommit"
+
+      Write-Host "##vso[task.prependpath]$TargetPath"
+      Write-Host "##vso[task.setvariable variable=VCPKG_INSTALLATION_ROOT]$TargetPath"
+      Write-Host "##vso[task.setvariable variable=VcpkgCommit]$VcpkgCommit"
+    displayName: Set Vcpkg Variables
     condition: >-
       and(
         succeeded(),
-        not(eq(variables['${{ parameters.DependenciesVariableName }}'], '')),
-        contains(variables['OSVmImage'], 'macOS')
+        not(eq(variables['${{ parameters.DependenciesVariableName }}'], ''))
       )
 
-  # Set MacOS specific build environment features for vcpkg
-  - task: Bash@3
+  - task: Cache@2
     inputs:
-      targetType: inline
-      script: |
-        sudo xcode-select --switch /Applications/Xcode_11.3.1.app
-        echo "xcode path:"
-        sudo xcode-select --print-path
-
-        # Install gcc 9
-        brew install gcc@9
-        gcc --version
-    displayName: MacOS - Set tools versions
+      key: >-
+        $(Agent.JobName)
+        | "$(VcpkgCommit)"
+        | $(Agent.Os)
+        | $(${{ parameters.DependenciesVariableName }})
+        | v2
```

Comment:
What is the v2 for? Is it to avoid any existing cache entries from before?

Reply:
This is to provide uniqueness to the cache key to test something. We can remove this now.